bootstrap_cfn/stacks/ec2.json

```diff
@@ -8,11 +8,26 @@
         "SecurityGroupIngress": []
     }
   },
+  "SaltSG": {
+    "Type": "AWS::EC2::SecurityGroup",
+    "Properties": {
+        "VpcId" : { "Ref" : "VPC" },
+        "GroupDescription": "Salt Comms Security Group",
+        "SecurityGroupIngress": [
+                                    {
+                                        "IpProtocol": "tcp",
+                                        "FromPort": 4505,
+                                        "ToPort": 4506,
```

Comment:
So... the difficulty is with circular references. The `security_groups` key in the yaml just gets dumped into BaseHostSG and `SourceSecurityGroupId` can't reference itself. So the answer is it can't go in the yaml without updating the code to create security groups rather than security group rules.